docs/ref/models/fields.txt

```diff
@@ -2266,6 +2201,113 @@ accepted by :class:`ForeignKey`, plus one extra argument:
 See :doc:`One-to-one relationships </topics/db/examples/one_to_one>` for usage
 examples of ``OneToOneField``.
 
+.. _lazy-relationships:
+
+Lazy relationships
+------------------
+
+Lazy relationships allow you to reference models using their names as strings
+or to create recursive relationships. You can use strings as the first
+argument in any relationship field to reference models lazily. The following
+types of lazy references are available:
+
+.. _recursive-relationships:
+
+Recursive
+~~~~~~~~~
+
+To define a relationship where a model references itself, use ``"self"`` as
+the first argument of the relationship field::
+
+    from django.db import models
+
+
+    class Manufacturer(models.Model):
+        suppliers = models.ManyToManyField("self", symmetrical=False)
+
+When used in an :ref:`abstract model <abstract-base-classes>`, the recursive
+relationship will point to concrete model subclasses.
+
+Relative
+~~~~~~~~
+
+If you need to create a relationship on a model that has not yet been defined,
+you can use the name of the model instead of the model object itself::
+
+    from django.db import models
+
+
+    class Car(models.Model):
+        manufacturer = models.ForeignKey(
+            "Manufacturer",
+            on_delete=models.CASCADE,
+        )
+        # ...
+
+
+    class Manufacturer(models.Model):
+        # ...
+        pass
+
+Relationships defined this way on :ref:`abstract models
+<abstract-base-classes>` are resolved when the model is subclassed as a
+concrete model and are not relative to the abstract model's ``app_label``:
+
+.. code-block:: python
+    :caption: ``products/models.py``
+
+    from django.db import models
+
+
+    class AbstractCar(models.Model):
+        manufacturer = models.ForeignKey("Manufacturer", on_delete=models.CASCADE)
+
+        class Meta:
+            abstract = True
+
+.. code-block:: python
+    :caption: ``production/models.py``
+
+    from django.db import models
+    from products.models import AbstractCar
+
+
+    class Manufacturer(models.Model):
+        pass
+
+
+    class Car(AbstractCar):
+        pass
+
+
+    # Car.manufacturer will point to `production.Manufacturer` here.
+
+.. admonition:: Blueprint models with relative references
```

Comment:
Added this as per @charettes's [comment](https://github.com/django/django/pull/6178#discussion_r54976680).

Reply:
Thank you for working on this @cliff688! For me (non native speaker), reading "blueprint models" raises an eyebrow because I'm not really sure what it means. Perhaps:
```suggestion
.. admonition:: Reusable models with relative references
```

Reply:
Yeah. I think if we can remove or explain "blueprint" in the note itself, then I believe we will have made the meaning clearer. Currently, removing or replacing it with "reusable" does not say anything especially noteworthy of relative references.

Reply:
I see your point @cliff688, thanks for raising that.

One idea would be to provide a small example within this subsection to illustrate this admonition. This would mean that the admonition title gets bumped to a subsubsection (is it possible?) and we provide a concise example. What I don't like about this approach is that this part of the docs is about "references", i.e. study material when following the [Diataxis model](https://diataxis.fr/). According to this, examples and "recipes" should be in the How-To section instead, because those are practical.

So, in summary, I'm not 100% sure of the best course of action. I think that an acceptable middle ground may be the upgrading of the admonition to a entry on its own with the smallest (perhaps slightly incomplete) example along the commentary to explain what a blueprint model is? What do you think?

(In the mid term, one could evaluate a separated, detailed How-To and point to that from here, but I would say that is outside the scope of this ticket.)

Reply:
Thank you @nessita for link to Diataxis! I enjoyed reading it.

About the admonition, I'm not in favour of adding an example here or in the "How-to" docs. Here's the admonition as it currently stands:
```
.. admonition:: Reusable models with relative references

    Relative references allow the creation of reusable abstract models
    with relationships that can resolve to different implementations
    of the referenced models in various subclasses across different
    applications.
```
Firstly, adding an example might overcomplicate the explanation, especially as the use case is more niche and abstract than common.

Secondly, I think as it currently stands, the admonition is explained in a way that should be clear to the target audience, and for those familiar with the concepts mentioned in the admonition, further illustration should be unnecessary.

Reply:
@cliff688 Then maybe I misunderstood your previous comment. When you said:

> removing or replacing it with "reusable" does not say anything especially noteworthy of relative references.

I thought you were implying that we should add "something" to the current proposal to improve it. Could you expand on what did you mean please?

Reply:
Oh, I see 👍🏾 When I made that comment, the admonition read:
```
    Relative references enable the definition of reusable blueprint models with
    abstract classes, allowing relationships to resolve in concrete subclasses
    across different applications.
```
which after removing "blueprint" would not say anything that hadn't been said before (i.e. we already mentioned how relative references resolve in concrete model subclasses, and we already know that abstract models are reusable). However, I changed the admonition some moments afterwards (outdating the comment) to:
```
.. admonition:: Reusable models with relative references

    Relative references allow the creation of reusable abstract models
    with relationships that can resolve to different implementations
    of the referenced models in various subclasses across different
    applications.
```
which I think says something noteworthy about relative references in particular; namely, the bit about pointing to different implementations using the same relative reference. Also, with the new wording, we had dropped "blueprint".

Reply:
Amazing, thank you @cliff688. I believe we are in agreement then, we'll keep the admonition as is.

I'll give a final review soon and merge afterwards, great work!

Reply:
Thank you @nessita 🌻